src/App.tsx

```diff
@@ -1,26 +1,225 @@
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [processingIds, setProcessingIds] = useState<number[]>([]);
+
+  const [title, setTitle] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const inputRef = React.useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load todos'));
+  }, []);
+
+  useEffect(() => {
+    if (!tempTodo) {
+      inputRef.current?.focus();
+    }
+  }, [tempTodo]);
+
+  const activeTodos = todos.filter(todo => !todo.completed);
+  const completedTodos = todos.filter(todo => todo.completed);
+
+  let visibleTodos = todos;
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const normalizedTitle = title.trim();
+
+    if (!normalizedTitle) {
+      setErrorMessage('Title should not be empty');
+
+      return;
+    }
+
+    setTempTodo({
+      id: 0,
+      title: normalizedTitle,
+      completed: false,
+      userId: 0,
+    });
+
+    setErrorMessage('');
+
+    createTodo(normalizedTitle)
+      .then(todo => {
+        setTodos((prevTodos: Todo[]) => [...prevTodos, todo]);
+        setTitle('');
+      })
+      .catch(() => {
+        setErrorMessage('Unable to add a todo');
+      })
+      .finally(() => setTempTodo(null));
+  };
+
+  const handleDeletingTodo = (id: number) => {
+    setErrorMessage('');
+    setProcessingIds(prevIds => [...prevIds, id]);
+
+    return deleteTodo(id)
+      .then(() => {
+        setTodos(prevTodos => prevTodos.filter(todo => todo.id !== id));
+      })
+      .catch(() => setErrorMessage('Unable to delete a todo'))
+      .finally(() =>
+        setProcessingIds(prevIds => prevIds.filter(prevId => prevId !== id)),
+      );
+  };
+
+  const handleClearCompleted = () => {
+    completedTodos.forEach(todo => handleDeletingTodo(todo.id));
+  };
+
+  useEffect(() => {
+    if (!errorMessage) {
+      return;
+    }
+
+    const timerID = setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+
+    return () => clearTimeout(timerID);
+  }, [errorMessage]);
+
+  if (status === Filter.active) {
+    visibleTodos = activeTodos;
+  } else if (status === Filter.completed) {
+    visibleTodos = completedTodos;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {/* this button should have `active` class only if all todos are completed */}
```

Comment:
Remove comments, it's unused code (Fix all cases)
BTW `if all todos are completed` this condition is not met
Please implement it in third part of `todo_app` task